imap_processing/swapi/l2/swapi_l2.py

```diff
@@ -0,0 +1,105 @@
+"""SWAPI L2 processing module."""
+
+import logging
+
+import xarray as xr
+
+from imap_processing.cdf.imap_cdf_manager import ImapCdfAttributes
+from imap_processing.swapi.swapi_utils import SWAPIAPID
+
+logger = logging.getLogger(__name__)
+
+
+TIME_PER_BIN = 0.167  # seconds
+
+
+def swapi_l2(l1_dataset: xr.Dataset, data_version: str) -> xr.Dataset:
+    """
+    Produce science data to L2.
+
+    To process science data to L2, we need to:
+    - convert counts to rates. This is done by dividing the counts by the
+        TIME_PER_BIN time. TIME_PER_BIN is the exposure time per energy bin which is
+        obtained by dividing the time for one complete sweep
+        (12 s, coarse + fine sweep) by the total energy steps (72),
+        i.e., TIME_PER_BIN = 12/72 = 0.167 s. This will be constant.
+
+    - update uncertainty. Calculate new uncertainty value using
+        SWP_PCEM_ERR data from level one and divide by TIME_PER_BIN. Eg.
+            SWP_PCEM_UNC = SWP_PCEM_ERR / TIME_PER_BIN
+        Do the same for SCEM and COIN data.
+
+    Parameters
+    ----------
+    l1_dataset : xarray.Dataset
+        The L1 data input.
+    data_version : str
+        Version of the data product being created.
+
+    Returns
+    -------
+    data : xarray.Dataset
+        Processed data to L2.
+    """
+    apid = int(l1_dataset.attrs["Apid"])
+    # Right now, we don't process any other apid besides science packet
+    # to L2
+    if apid != SWAPIAPID.SWP_SCI:
+        raise ValueError(f"APID {apid} is not supported for L2 processing")
```

Comment:
Thinking about this more. Why do we have an apid check at level 2, shouldn't this actually be a "level 1 filename check" instead?

All this to say, I'm not actually sure we need this check, or if it really should be up in the CLI dependencies side of things when you are checking the length of dependencies you could also check the filename there if you wanted.

Reply:
True, this will be prevent by dependency length. Ok. I can remove it.